core/task/descriptor.go

```diff
@@ -48,6 +49,8 @@ type Descriptor struct {
 	TaskRole          parentRole
 	TaskClassName     string
 	RoleConstraints   constraint.Constraints
+	RoleConnect       []channel.Outbound
+	RoleInbound       []channel.Inbound
```

Comment:
Do we need these 2 in the `Descriptor` because we need to be able to access the full role hierarchy to do a merge from `scheduler.go`?

Reply:
Yeah exactly.